tests/unit/test_index_html_page.py

```diff
@@ -0,0 +1,162 @@
+import logging
+
+import mock
+import pytest
+from pip._vendor.six.moves.urllib import request as urllib_request
+
+from pip._internal.download import PipSession
+from pip._internal.index import (
+    Link, _get_html_page, _get_html_response, _NotHTML, _NotHTTP,
+)
+
+
+@pytest.mark.parametrize(
+    "url",
+    [
+        "ftp://python.org/python-3.7.1.zip",
+        "file:///opt/data/pip-18.0.tar.gz",
+    ],
+)
+def test_get_html_response_archive_to_naive_scheme(url):
+    """
+    `_get_html_response()` should error on an archive-like URL if the scheme
+    does not allow "poking" without getting data.
+    """
+    with pytest.raises(_NotHTTP):
+        _get_html_response(url, session=mock.Mock(PipSession))
+
+
+@pytest.mark.parametrize(
+    "url, content_type",
+    [
+        ("http://python.org/python-3.7.1.zip", "application/zip"),
+        ("https://pypi.org/pip-18.0.tar.gz", "application/gzip"),
+    ],
+)
+def test_get_html_response_archive_to_http_scheme(url, content_type):
+    """
+    `_get_html_response()` should send a HEAD request on an archive-like URL
+    if the scheme supports it, and raise `_NotHTML` if the response isn't HTML.
+    """
+    session = mock.Mock(PipSession)
+    session.head.return_value = mock.Mock(**{
+        "request.method": "HEAD",
+        "headers": {"Content-Type": content_type},
+    })
+
+    with pytest.raises(_NotHTML) as ctx:
+        _get_html_response(url, session=session)
+
+    session.assert_has_calls([
+        mock.call.head(url, allow_redirects=True),
+    ])
+    assert ctx.value.args == (content_type, "HEAD")
+
+
+@pytest.mark.parametrize(
+    "url",
+    [
+        "http://python.org/python-3.7.1.zip",
+        "https://pypi.org/pip-18.0.tar.gz",
+    ],
+)
+def test_get_html_response_archive_to_http_scheme_is_html(url):
+    """
+    `_get_html_response()` should work with archive-like URLs if the HEAD
+    request is responded by text/html.
```

Comment:
"request responds with text/html."

Reply:
Should this be “… request is responded with text/html” instead?

Reply:
For that form I would say, "request is responded to with text/html."

But even simpler is "if the response to the HEAD request is text/html" or even "if the HEAD request response is text/html."